src/status_im/data_store/browser.cljs

```diff
@@ -1,26 +1,49 @@
 (ns status-im.data-store.browser
   (:require [re-frame.core :as re-frame]
-            [status-im.data-store.realm.core :as core]))
+            [status-im.utils.fx :as fx]
+            [status-im.ethereum.json-rpc :as json-rpc]))
 
-(re-frame/reg-cofx
- :data-store/all-browsers
- (fn [cofx _]
-   (assoc cofx :all-stored-browsers (-> @core/account-realm
-                                        (core/get-all :browser)
-                                        (core/sorted :timestamp :desc)
-                                        (core/all-clj :browser)))))
+;; TODO: adjust the api so there is no need to do these transformations after
+;; benchmarking the cost of recreating all these maps with different keys
+(fx/defn initialize-browsers
+  {:events [::browsers]}
+  [{:keys [db]} all-stored-browsers]
+  (let [browsers (reduce (fn [acc {:keys [id name timestamp dapp historyIndex history]}]
+                           (assoc acc id {:browser-id id
+                                          :name name
+                                          :timestamp timestamp
+                                          :dapp dapp
+                                          :history-index historyIndex
+                                          :history history}))
+                         {}
+                         all-stored-browsers)]
+    {:db (assoc db :browser/browsers browsers)}))
+
+(re-frame/reg-fx
+ :data-store/get-browsers
+ (fn []
+   (json-rpc/call
+    {:method "browsers_getBrowsers"
+     :on-success #(re-frame/dispatch [::browsers %])})))
```

Comment:
why not `:browsers/initialize-browsers` ?

Reply:
`::initialize-browser` maybe? Since we are using fx/defn now to declare event, I think we should start to use `::` more because then we can do aliased namespaced keywords instead of synthetic ones, and those can be found more easily accross the code base by IDEs

Reply:
true